### PR TITLE
added security first part with User class

### PR DESCRIPTION
**Pull request by yuriy-kononchuk**

### CODE REVIEW

File: src/main/java/com/example/project/dto/user/UserRegistrationRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.example.project.dto.user;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
```
Comment (okuzan) [inline src/main/java/com/example/project/dto/user/UserRegistrationRequestDto.java]: use jakarta.validation.constraints.Size;

File: src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -56,4 +56,17 @@ protected ResponseEntity<Object> handleEntityNotFoundException(
         bodyToResponce.put("message", ex.getMessage());
         return new ResponseEntity<>(bodyToResponce, HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(value = {RegistrationException.class})
+    protected ResponseEntity<Object> handlRegistrationException(
```
Comment (okuzan) [inline src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java]: handl**e**

File: src/main/java/com/example/project/mapper/UserMapperImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        dto.setId(user.getId());
+        dto.setEmail(user.getEmail());
+        dto.setFirstName(user.getFirstName());
+        dto.setLastName(user.getLastName());
+        dto.setShippingAddress(user.getShippingAddress());
+        return dto;
+    }
+
+    @Override
+    public User convertToModel(UserRegistrationRequestDto userDto) {
```
Comment (okuzan) [inline src/main/java/com/example/project/mapper/UserMapperImpl.java]: toModel
unify method naming :)

**Review by okuzan — APPROVED** [review]

File: src/main/java/com/example/project/dto/user/UserRegistrationRequestDto.java (hunk 1 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/dto/user/UserRegistrationRequestDto.java]: got it

File: src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java (hunk 2 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java]: exactly

File: src/main/java/com/example/project/mapper/UserMapperImpl.java (hunk 3 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/mapper/UserMapperImpl.java]: got it

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**